crates/criticalup-core/src/download_server_client.rs

```diff
@@ -368,6 +373,18 @@ mod tests {
         assert_eq!(req.headers().get(IF_NONE_MATCH), Some(&test_hash));
     }
 
+    #[tokio::test]
+    async fn cache_is_constructed() {
+        let test_env = TestEnvironment::with().download_server().prepare().await;
+
+        let res = test_env
+            .download_server()
+            .product_release_cache_path("ferrocene", "stable-25.05.0");
+        let cache_dir: PathBuf = test_env.config().paths.cache_dir.clone();
+        let expected = "artifacts/products/ferrocene/releases/stable-25.05.0";
+        let cache_dir = cache_dir.join(expected);
```

Comment:
Ah lovely, this is what I was trying to do, but I didn't do the `join()` method. So it didn't work :(
I will remember this!